nrf70_bm_lib/source/system/nrf70_bm_lib.c

```diff
@@ -187,18 +187,14 @@ int nrf70_bm_sys_deinit(void)
 	ret = nrf70_bm_sys_fmac_del_vif_sta();
 	if (ret) {
 		NRF70_LOG_ERR("Failed to delete STA VIF");
-		goto err;
 	}
 
 	// Clean up the WiFi module
 	ret = nrf70_bm_sys_fmac_deinit();
 	if (ret) {
 		NRF70_LOG_ERR("Failed to deinitialize FMAC module");
-		goto err;
 	}
 
-	return 0;
-err:
 	return ret;
```

Comment:
Should the function return a `fail` status in case one of the steps fails? Right now (hypothetically) if  `nrf70_bm_sys_fmac_deinit` passes after `nrf70_bm_sys_fmac_del_vif_sta` has failed then the function will return a `PASS`.

Reply:
It was intentional, what or how will the user deal with the intermediate failures? If all fail then we return failure, but as long as the last step passes, IMHO user should not be aware and for developers we log every failure.